File: OOPs/_2_Encapsulation/encapsulation.dart
class Animal {
  String _name;  // Private attribute (starts with an underscore)
  
  Animal(this._name);
  
  // Getter
  String get name => _name;
  
  // Setter
  set name(String newName) {
    _name = newName;
  }
}

void main() {
  Animal cat = Animal('Cat');
  print(cat.name);  // Cat
  cat.name = 'Lion';
  print(cat.name);  // Lion
}

File: OOPs/_2_Encapsulation/encapsulation.py
class Animal:
    def __init__(self, name):
        self.__name = name  # Private attribute (two underscores)
    
    # Getter
    @property
    def name(self):
        return self.__name
    
    # Setter
    @name.setter
    def name(self, new_name):
        self.__name = new_name

cat = Animal('Cat')
print(cat.name)  # Cat
cat.name = 'Lion'
print(cat.name)  # Lion
